Add Machine Learning feature flag (#3351)
testcontainers/testcontainers-java: docs/modules/elasticsearch.md

(not recommended as it is deprecated), the default cluster name is set to `docker-cluster` so you need to change `cluster.name` setting
or set `client.transport.ignore_cluster_name` to `true`.

## Enable / Disable features

The default distribution of Elasticsearch comes with the basic Elastic license which contains a lot of features.

> You can turn on and off those features.

Reviewer: ```suggestion
You can turn those features on and off.
```

[HttpClient](../../modules/elasticsearch/src/test/java/org/testcontainers/elasticsearch/ElasticsearchContainerTest.java) inside_block:httpClientSecuredContainer
<!--/codeinclude-->

### Machine Learning

> The Machine Learning feature is rarely used within integration tests. The Elasticsearch Container module disables
> by default this feature. You can activate it by calling `withMachineLearning()` when building the container:

Reviewer: ```suggestion
The Machine Learning feature is rarely used within integration tests. 
The Elasticsearch Container module disables it by default.
However, you can activate it if needed:
```
